pkg/mcs/resource_manager/client/client.go

```diff
@@ -535,7 +535,6 @@ func (gc *groupCostController) handleTokenBucketTrickEvent(ctx context.Context,
 				threshold := counter.notify.setupNotificationThreshold
 				counter.notify.mu.Unlock()
 				counter.limiter.SetupNotificationThreshold(now, threshold)
-				gc.updateRunState(ctx)
```

Comment:
Why removed this line?

Reply:
https://github.com/tikv/pd/pull/5891 introduced this race bug which will have the WAW problem with other `updateRunState` of `mainLoop`.